pkg/guestagent/guestagent_linux.go

```diff
@@ -244,3 +246,30 @@ func (a *agent) Info(ctx context.Context) (*api.Info, error) {
 	}
 	return &info, nil
 }
+
+const deltaLimit = time.Second * 5
+
+func (a *agent) fixSystemTimeSkew() {
+	ticker := time.NewTicker(time.Second * 10)
+	defer ticker.Stop()
+	for now := range ticker.C {
+		rtc, err := timesync.GetRTCTime()
+		if err != nil {
+			logrus.Warnf("fixSystemTimeSkew: lookup error: %s", err.Error())
+			continue
+		}
+		d := rtc.Sub(now)
+		logrus.Debugf("fixSystemTimeSkew: rtc=%s systime=%s delta=%s",
+			rtc.Format(time.RFC3339), now.Format(time.RFC3339), d)
+		if (d > 0 && d > deltaLimit) || (d < 0 && d < -deltaLimit) {
```

Comment:
Aren't the comparisons with `0` redundant?

```suggestion
		if d > deltaLimit || d < -deltaLimit {
```

I'm also slightly concerned about adjusting the time backwards, as that can create subtle problems, but in practice I can only see the guest time lagging behind the host time, so it doesn't really matter.